waspc/packages/ts-inspect/src/exports.ts

```diff
@@ -0,0 +1,77 @@
+import ts from 'typescript';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import JSON5 from 'json5';
+
+export type Export
+  = { type: 'default' } & Location
+  | { type: 'named', name: string } & Location
+
+export type Location = { location?: { line: number, column: number } }
+
+export async function getExportsOfFiles(
+  filenames: string[],
+  tsconfigFilename?: string,
+): Promise<{ [file: string]: Export[] }> {
+  let compilerOptions: ts.CompilerOptions = {};
+
+  // If a tsconfig is given, load the configuration.
+  if (tsconfigFilename) {
+    const configJson = JSON5.parse(await fs.readFile(tsconfigFilename, 'utf8'));
+    const basePath = path.dirname(tsconfigFilename)
+
+    const { options, errors } = ts.convertCompilerOptionsFromJson(
+      configJson.compilerOptions, basePath, tsconfigFilename
+    );
+    if (errors && errors.length) {
+      throw errors;
+    }
+    compilerOptions = options;
+  }
+
+  const exportsMap: { [file: string]: Export[] } = {};
+
+  // Initialize the TS compiler.
+  const program = ts.createProgram(filenames, compilerOptions);
+  const checker = program.getTypeChecker();
+
+  // Loop through each given file and try to get its exports.
+  for (let filename of filenames) {
+    const source = program.getSourceFile(filename);
```

Comment:
I would maybe extract the body of the loop into a helper function `getExportsForFile` and maybe do some `throw`ing and `catch`ing in the loop to output the errors and continue the loop instead of using `continue`